### PR TITLE
Make DEFAULT_LIMIT monkey patchable

### PR DESCRIPTION
**Pull request by spacemanspiff2007**
## What do these changes do?

Currently its not possible to change DEFAULT_LIMIT.
This change allows this:
```
import aiohttp
aiohttp.streams.DEFAULT_LIMIT = 10 * aiohttp.streams.DEFAULT_LIMIT
```

## Are there changes in behavior for the user?
No

## Related issue number
https://github.com/aio-libs/aiohttp/issues/4453

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by spacemanspiff2007** [comment]
@webknjaz could you review this change, too? It would fix my application from crashing.

File: aiohttp/streams.py [hunk 1]
```diff
@@ -601,12 +601,12 @@ class FlowControlDataQueue(DataQueue[_T]):
...
 
     def __init__(self, protocol: BaseProtocol, *,
-                 limit: int=DEFAULT_LIMIT,
+                 limit: Optional[int]=None,
                  loop: asyncio.AbstractEventLoop) -> None:
         super().__init__(loop=loop)
 
         self._protocol = protocol
-        self._limit = limit * 2
+        self._limit = limit * 2 if limit is not None else DEFAULT_LIMIT
```
Comment (webknjaz) [inline aiohttp/streams.py]: this makes the default limit 2 times smaller

**Comment by spacemanspiff2007** [comment]
I changed it as requested - it should be possible to merge this.

**Comment by spacemanspiff2007** [comment]
Is there anything I can do to get this merged?

**Comment by robhaswell** [comment]
I am another user who would like to see this functionality.

**Review by asvetlov — CHANGES_REQUESTED** [review]
Sorry, I don't support this PR.
If you want to change the default buffer size -- please pass the desired value into `client.get()` or a similar call instead of monkey-patching. `ClientSession` can accept the session-default value as the argument constructor as well.

Sure, it requires a bigger pull request than initially proposed but I believe that explicit way is much better.

**Comment by spacemanspiff2007** [comment]
Thank you for your reply. The crash happens when using aiohttp together with [aiohttp-sse-client](https://github.com/rtfol/aiohttp-sse-client) but I think if it's a possible Session arg this would be still fine.
A working example and finished PR would be [here](https://github.com/aio-libs/aiohttp/pull/4456), it just has to be merged.

I am just looking for any (simple) way to change the buffer so my application doesn't go down with an exception

**Comment by asvetlov** [comment]
@spacemanspiff2007 sorry again, I misled you and myself by writing controversial recommendations.

To don't bother you again I've prepared a fix myself as #5065
I hope it solves your needs.

**Comment by asvetlov** [comment]
Closing the PR for the sake of #5056

**Comment by spacemanspiff2007** [comment]
Thank you very much for your efforts and for the solution!

**Comment by robhaswell** [comment]
Sorry @asvetlov I must be being dumb, but I can't see how #5056 is related to the size of the line buffer?

**Comment by asvetlov** [comment]
Sorry for the typo, the actual number is #5065 instead of #5056

**Comment by robhaswell** [comment]
Got it. Thanks for addressing this issue.

**Pull request closed without merging**